### PR TITLE
Workaround for empty restore.txt file under .tam

### PR DESCRIPTION
**Pull request by livarcocc**
For some reason, specially in CI, the restore.txt file under .tam is ending up empty. As a workaround for it, I am going to retry the restore if that happens.

This is an attempt as a work around. Let's see how CI behaves with it.

@piotrpMSFT @krwq @jgoshi

### CODE REVIEW

File: src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs [hunk 1]
```diff
@@ -141,14 +141,31 @@ internal IEnumerable<FileInfo> GetRestoreFiles()
...
+            Action action)
         {
+            var inventory = Enumerable.Empty<FileInfo>();
             if (file.Exists)
             {
-                return LoadInventory(file);
+                inventory = LoadInventory(file);
+            }
+
+            if(inventory.Any())
```
Comment (piotrpMSFT) [inline src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs]: Why? the file being empty is a perfectly valid state.

File: src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs [hunk 2]
```diff
@@ -141,14 +141,31 @@ internal IEnumerable<FileInfo> GetRestoreFiles()
 
         internal IEnumerable<FileInfo> GetBuildFiles()
         {
-            return GetInventory(_inventoryFiles.Build, GetRestoreFiles, DoBuild);
+            return GetInventory(
+                _inventoryFiles.Build,
+                () =>
+                {
+                    var preInventory = new List<FileInfo>(GetRestoreFiles());
```
Comment (piotrpMSFT) [inline src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs]: Why is this needed? The previous logic would just take a before/after snapshot and diff. It didn't take dependencies on the order of events for the calculation. Only the execution of a command caused its predecessor to be executed. I don't like the duplication of knowledge.

File: src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs (hunk 1 above)
Comment (livarcocc) [inline src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs]: How so? That's exactly the problem, the restore file is being empty when it shouldn't.

Comment (piotrpMSFT) [inline src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs]: The problem is not that the restore file is empty and shouldn't be. That's a consequence. The problem is that restore failed and we didn't fail execution overall. If restore fails, that file should not be written.

Comment (livarcocc) [inline src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs]: There is no evidence that restore failed. I reproed this a couple of times here on my box and restore did not seem to fail.

**Comment by piotrpMSFT** [comment]
@MattGertz

**Comment by livarcocc** [comment]
@MattGertz Can you approve this?
